### PR TITLE
REALM-8021: handle hex keys for objects

### PR DESCRIPTION
**Pull request by Gabri3l**

### CODE REVIEW

File: object_test.go [hunk 1]
```diff
@@ -103,6 +103,12 @@ func TestObject_create(t *testing.T) {
             var mno = Object.create({}, properties);
             mno.hasOwnProperty("abc");
         `, false)
+
+		test(`
+		    var myO = { 0x16: "hexciting" };
+
+		    myO[22]
+		`, "hexciting")
```
[the remark this answers is not in the document]
Comment (mburdette) [reply object_test.go:111]: [nit] can we get a test where this would leave `"0x16"` as that string?

**Review by mburdette — APPROVED** [review]
LGTM 😎 

[commit/pr nit] can we make the main commit/PR title actually reflect work done here and not a restating of the issue? example `convert hexadecimal keys to base 10 numbers in...`

Just the one nit to add a test.

File: object_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (Gabri3l) [reply object_test.go]: I'm not sure I understand but I updated the test to also return `myO[0x16]` to make sure that if we use hex as a key it all works as expected.

[the remark this answers is not in the document]
Comment (mburdette) [reply object_test.go]: like, we need a test to make sure that your fix doesn't break the string case

**Comment by Gabri3l** [comment]
I found a better place where this change should happen, which does NOT affect any other type of get/set with object keys that are not numerical.

**Pull request merged**